### PR TITLE
Support waas 1.9

### LINKED ISSUE #73: Support WaaS 1.9
https://github.com/Tangany/waas/releases/tag/v1.9.0

### PR DESCRIPTION
**Pull request by lenabl**
- Extend wallet information for btc and eth
- Add "amount" as optional parameter for btc transactions, add optional sweep parameter for fee estimation
- Extend transaction estimation for btc and eth with new properties
- Add Tezos functionality
- Add tests for Tezos integration
- Remove all deprecated methods for eth and btc
- Update eth and btc tests

Resolves #73

### CODE REVIEW

File: src/xtz-wallet.spec.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    });
+
+    describe("delegateAsync", function() {
+        it("should execute the api call", async function() {
+            const sampleRequestId = "71c4f385a4124239b6c968e47ea95f73";
+            const postStub = this.waas.instance.post = this.sandbox.stub().resolves({statusUri: `request/${sampleRequestId}`, id: `${sampleRequestId}`});
+            const wallet = new Wallet(this.waas, sampleWallet);
+            const tezosWallet = new TezosWallet(this.waas, wallet);
+            const request = await tezosWallet.delegateAsync(sampleAddress);
+            assert.strictEqual(postStub.withArgs(`${baseUrl}/delegate-async`, sampleAddress).callCount, 1);
```
Comment (kepikoi) [inline src/xtz-wallet.spec.ts:67]: My unit test run is failing at this line. Can you please check this?

File: test/tezos.e2e.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+	it("should send XTZ to other test wallet asynchronously", async function () {
+		const request = await api.wallet(wallet).xtz().sendAsync(recipients);
+		console.log(`Started request with id ${request.id} ...`);
+		const { process } = await request.get();
+		assert.ok(process);
+		console.log(`The transaction process is ${process}.`)
+	});
+
+	it("should cancel current delegate", async function () {
```
Comment (kepikoi) [inline test/tezos.e2e.js:114]: I would rather _set_ an new delegate on the temporary wallet created in the second unit to retain some degree of idempotency in the E2E test

File: src/xtz-contract-wallet.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    constructor(waas: Waas, walletInstance: Wallet, public readonly address: string) {
+        super(waas, walletInstance);
+        t("String", address);
+        this.baseUrl = `xtz/contract/${address}/${this.wallet}`;
+    }
+
+    /**
+     * Executes methods of arbitrary Tezos smart contracts
+     * @param config Smart contract configuration
```
Comment (kepikoi) [inline src/xtz-contract-wallet.ts:22]:
```suggestion
     * @param config - Smart contract configuration
```

File: src/interfaces/tezos.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+export interface IXtzNodeInfo {
+    protocol: string;
+    blockHeight: number;
+    chainId: string;
+}
+
+export interface IXtzStatus extends INodeStatus<IXtzNodeInfo> {
+}
+
+export interface ITezosOperationStatusOutput {
```
Comment (kepikoi) [inline src/interfaces/tezos.ts:13]: Please describe the interfaces via JSDoc annotations (compare ethereum.ts or bitcoin.ts)

File: test/bitcoin.e2e.js [hunk 5]
```diff
@@ -88,24 +90,10 @@ describe("WaaS sample Bitcoin workflow", function () {
...
 		const req = await noConfirmationsBtcApi.wallet(createdWallet).btc().sweepAsync({wallet: wallet});
 		await req.wait(35e3);
 		const { output } = await req.get();
 		console.log(output);
 	})
-
-	it("should create a signed transaction that can be manually transmitted", async function () {
-		const { rawTransaction } = await noConfirmationsBtcApi.wallet(wallet).btc().sign(recipients);
-		console.log(`signing endpoint returned '${rawTransaction}'`);
-	});
```
Comment (kepikoi) [inline test/bitcoin.e2e.js:110]: Please add a sweep estimation test to the E2E suite

File: src/waas.ts [hunk 6]
```diff
@@ -75,6 +80,7 @@ interface IWaaSOptions {
     bitcoinTxConfirmations?: BlockchainTxConfirmations | number;
     bitcoinTxSpeed?: BitcoinTxSpeed;
     bitcoinMaxFeeRate?: number;
+    tezosNetwork?: TezosNetwork;
 }
```
Comment (kepikoi) [inline src/waas.ts:84]: Please add the missing Tezos header configuration options: `tangany-tezos-gas-limit`, `tangany-tezos-storage-limit`, `tangany-tezos-operation-confirmations`. 
Also add those as module exports to `src/index.ts`

**Review by kepikoi — CHANGES_REQUESTED** [review]
- Please update the change log (under ## [Unreleased])
- Update the `Available configuration headers` in `readme.md` with the Tezos configuration options (`tezosOperationConfirmations` etc.)
- Add Tezos code samples below `### More examples` to `readme.md` 
- Remove the deprecated (and meanwhile decommissioned) Ethereum ERC20 methods `send`, `approve`, `transferFrom`, `burn`, and `mint` as well as the dependent `getRecipientsData` method

**Comment by lenabl** [comment]
Made the requested changes:

* Fix failing unit test
* Add new e2e tests
* Add missing headers
* Remove deprecated methods
* Update readme and changelog

File: src/xtz-wallet.spec.ts (hunk 1 above)
Comment (lenabl) [reply src/xtz-wallet.spec.ts]: Resolved in 0254db89c7fbdcd62a3e8a12cb6470cb03e434eb

File: test/tezos.e2e.js (hunk 2 above)
Comment (lenabl) [reply test/tezos.e2e.js:123]: Resolved in eac9da2fbd567b7f5e1fea254544805c2d0788a7

File: src/interfaces/tezos.ts (hunk 4 above)
Comment (lenabl) [reply src/interfaces/tezos.ts:19]: Resolved in e8c2a5a001829670d9d83c62500b64541fea6bff

File: test/bitcoin.e2e.js (hunk 5 above)
Comment (lenabl) [reply test/bitcoin.e2e.js:110]: Resolved in 5499b30a882e75be3226526c0dc8520fac4240ed

File: src/waas.ts (hunk 6 above)
Comment (lenabl) [reply src/waas.ts:87]: Resolved in 3b12e580d9510745b14c2cf68fe2894cffa5ea1c

**Review by kepikoi — APPROVED** [review]

**Pull request merged**